link-grammar/dict-common/print-dict.c

```diff
@@ -247,17 +246,18 @@ static char *display_word_split(Dictionary dict,
 
 	dyn_str *s = dyn_str_new();
 
-	parse_options_set_spell_guess(&display_word_opts, 0);
+	int spell_option = parse_options_get_spell_guess(opts);
+	parse_options_set_spell_guess(opts, 0);
 	sent = sentence_create(pword, dict);
-	if (0 == sentence_split(sent, &display_word_opts))
+	if (0 == sentence_split(sent, opts))
 	{
 		/* List the splits */
 		print_sentence_word_alternatives(s, sent, false, NULL, NULL);
 		/* List the disjuncts information. */
 		print_sentence_word_alternatives(s, sent, false, display, NULL);
 	}
 	sentence_delete(sent);
-	free_cost_table(&display_word_opts); /* XXX */
```

Comment:
It would have been enough to remove this one line.  The struct copy does not copy things that get pointed at.